compiler/lib/parse_bytecode.ml

```diff
@@ -258,31 +258,21 @@ end = struct
       |> List.sort ~cmp:(fun (i, _) (j, _) -> compare i j)
     with Not_found -> []
 
-  let mem { events_by_pc; _ } = Int_table.mem events_by_pc
+  let mem { events_by_pc; _ } pc = Int_table.mem events_by_pc pc
 
   let find_loc { events_by_pc; _ } ?(after = false) pc =
     try
-      let before, { event; source } =
-        try false, Int_table.find events_by_pc pc
-        with Not_found -> (
-          ( true
-          , try Int_table.find events_by_pc (pc + 1)
-            with Not_found -> (
-              try Int_table.find events_by_pc (pc + 2)
-              with Not_found -> Int_table.find events_by_pc (pc + 3)) ))
-      in
+      let { event; source } = Int_table.find events_by_pc pc in
       let loc = event.ev_loc in
       if loc.Location.loc_ghost
       then None
       else
         let pos =
           if after
           then loc.Location.loc_end
-          else if before
-          then loc.Location.loc_start
           else
             match event.ev_kind with
-            | Event_after _ -> loc.Location.loc_end
+            | Event_after _ when false -> loc.Location.loc_end
```

Comment:
For an event that is after an expression, the relevant location should be at the end of the expression, shouldn't it?